swarm.go

```diff
@@ -392,15 +393,17 @@ func (s *Swarm) ConnsToPeer(p peer.ID) []network.Conn {
 
 // bestConnToPeer returns the best connection to peer.
 func (s *Swarm) bestConnToPeer(p peer.ID) *Conn {
-	// Selects the best connection we have to the peer.
-	// TODO: Prefer some transports over others. Currently, we just select
-	// the newest non-closed connection with the most streams.
+
+	// TODO: Prefer some transports over others.
+	// For now, prefers direct connections over Relayed connections.
+	// For tie-breaking, select the newest non-closed connection with the most streams.
 	s.conns.RLock()
 	defer s.conns.RUnlock()
 
 	var best *Conn
 	bestLen := 0
-	for _, c := range s.conns.m[p] {
+	for i := range s.conns.m[p] {
```

Comment:
so we don't use the `i` anywhere, why not write this as `for _, c := ...` as it was before?

Reply:
Let me know if I'm wrong but for a slice of reference types(pointers), the reference semantic form of the for loop works better than the value semantic form ?

Reply:
Given that we are looping over a collection of reference types(pointers), it's easier to reason about the code if we use the reference semantic form of the range loop and avoid copies:

https://www.ardanlabs.com/blog/2017/06/for-range-semantics.html

Reply:
this is nonsense; we don't use the i and we iterate over connections. 
We should use the builtin syntax for that, which turns one line into two.

Reply:
done.